tests/test1-rewrite-2/task1/task1.c

```diff
@@ -0,0 +1,67 @@
+﻿#define _CRT_SECURE_NO_WARNINGS
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+bool isPalindrome(char* string, int stringLength) {
+    if (strcmp(string, "") == 0) {
+        return true;
+    }
+    for (int i = 0; i < stringLength / 2; ++i) {
+        if (string[i] != string[stringLength - i - 1]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool test1() {
+    char string[6] = "abccba";
+    return isPalindrome(string, 6);
+}
+
+bool test2() {
+    char string[6] = "abcbca";
+    return !isPalindrome(string, 6);
+}
+
+bool test3() {
+    char* string = "";
+    return isPalindrome(string, 0);
+}
+
+
+void main() {
+    if (!(test1() && test2() && test3())) {
+        printf("Test problems");
+        return;
+    }
+    int stringLength = 0;
+    int scanResult = 0;
+    while (!scanResult && stringLength <= 0) {
```

Comment:
```suggestion
    while (scanResult == 0 && stringLength <= 0) {
```
Лучше не пользоваться неявным кастом, в современных языках он обычно вообще запрещён